docs/knt-backend-doc.yml

```diff
@@ -0,0 +1,363 @@
+      parameters: []
+      responses:
+        '200':
+          description: A list of basic user objects with removed sensitive information
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/user"
+                  
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /products:
+    get:
+      tags:
+        - "Product Basic"
+      description: 'Retrieve a list of products from the database'
+      parameters: []
+      security:
+        - ApiKeyAuth: []
+      responses:
+        '200':
+          description: A list of product objects
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/product"
+                  
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /admin/products:
+    post:
+      tags:
+        - "Product Admin"
+      description: 'Create a new product in the database'
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        description: 'Basic product to create'
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/product'
+
+      responses:
+        '201':
+          description: "Created"
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /users/{userId}/purchase:
+    post:
+      tags:
+        - "User Basic"
+      description: 'Send purchase ammounts'
+      security:
+        - ApiKeyAuth: []
+      parameters: 
+        - in: path
+          name: userId
+          schema:
+            type: integer
+          required: true
+          description: Numeric ID of the user to target
+      requestBody:
+        description: 'List of products the user has bought'
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                type: object
+                properties:
+                  id:
+                    type: integer
+                    description: id of the product
+                  amount:
+                    type: integer
+                    description: amount of the product
+      responses:
+        '201':
+          description: "Created"
+        '401':
+          $ref: "#/components/responses/UnauthorizedError"
+        '407':
+          $ref: "#/components/responses/ForbidenError"
+  /users/{userId}:
+    get:
+      tags:
+        - "User Basic"
+      description: 'Get basic data of a specific user'
```

Comment:
When you say basic data you most likely want to define beforehand what you mean by it, in our case something like "Get data of a specific user, chosen by ID, without sensitive information" can work better. Again, it's just wording suggestions, I might revise it myself later, but I might as well mention it so that I would not forget.

Reply:
I defined the basic user scheme twice. Both in the request and in the global schema list, is that not enough?

Reply:
I changed the wording a bit to not distinguish user and "full user"

Reply:
yes, that's exactly what I mean